community/kernel/src/main/java/org/neo4j/kernel/impl/proc/ReflectiveProcedureCompiler.java

```diff
@@ -175,7 +175,18 @@
             ArrayList<CallableProcedure> out = new ArrayList<>( procedureMethods.size() );
             for ( Method method : procedureMethods )
             {
-                out.add( compileProcedure( procDefinition, constructor, method, warning, fullAccess ) );
+                String valueName = method.getAnnotation( Procedure.class ).value();
+                String definedName = method.getAnnotation( Procedure.class ).name();
+                QualifiedName procName = extractName( procDefinition, method, valueName, definedName );
+
+                if ( !(fullAccess || config.whiteListed( procName.toString() )) )
```

Comment:
Maybe switch order of the if and else part of this?
`fullAccess || config.whiteListed( procName.toString()` is easier to read and understand